lib/src/http/common/request.dart

```diff
@@ -707,4 +704,60 @@ abstract class CommonRequest extends Object
     didSucceed = true;
     return response;
   }
+
+  int _baseExponentialBackOffFunction(int numAttempts) {
+    return autoRetry.backOff.interval.inMilliseconds *
+        pow(defaultExponentialMultiplier, numAttempts);
+  }
+
+  Duration _createExponentialBackOff() {
+    Duration backOff;
+    Random random = new Random();
+
+    if (autoRetry.backOff.withJitter != null &&
+        autoRetry.backOff.withJitter == true) {
+      backOff = new Duration(
+          milliseconds: random.nextInt(min(
+              autoRetry.backOff.maxInterval.inMilliseconds,
+              _baseExponentialBackOffFunction(autoRetry.numAttempts))));
+    } else {
+      backOff = new Duration(
+          milliseconds: min(autoRetry.backOff.maxInterval.inMilliseconds,
+              _baseExponentialBackOffFunction(autoRetry.numAttempts)));
+    }
+
+    return backOff;
+  }
```

Comment:
replace `_baseExponentialBackOffFunction ` and `_createExponentialBackOff` with something like this?
```
  Duration _createExponentialBackOff() {
    int backOffInMs = autoRetry.backOff.interval.inMilliseconds *
        pow(defaultExponentialMultiplier, autoRetry.numAttempts);
    backOffInMs = min(autoRetry.backOff.maxInterval.inMilliseconds, backOffInMs);

    if (autoRetry.backOff.withJitter != null &&
        autoRetry.backOff.withJitter == true) {
      Random random = new Random();
      backOffInMs = random.nextInt(backOffInMs);
    }
    return new Duration(milliseconds: backOffInMs);
  }
```